### DISCUSSION TITLE
transaction for mixed insert on database and storage

### DISCUSSION THREAD

**Discussion by vikyw89** [supabase/supabase#14150] [Questions] [answered]
What's the best practice when I want to have data that is mixed in nature, blob and url stored in a single transaction ?

Example: I want to store title, name, description, etc etc as text in table or tables, which i can do using RPC from client side. But how do I store those things plus let's say a video, or an image which will be stored in storage ? Can I do RPC for storage not just table ?

**Comment by GaryAustin1** [comment, marked as answer]
Although technically you can call the storage-api from within a postgres function with http or pg_net extension, that seems inefficient (two transfers of your blob data) versus just doing the storage call and database call and dealing with error of either in client code.  And edge function might also be a could use for this.
